### PR TITLE
Enhancing AthenaSQLStage to handle both static and dynamic SQL Queries

### PR DESCRIPTION
**Pull request by anmolsgandhi**
### Enhancement
AthenaSQLStage to handle both static and dynamic SQL Queries

### Detail
Stage Enhanced to support both Static Queries (current case) and dynamic queries using statemachine json path to add the ability to run multiple queries using the same state machine

### Relates
(core): Athena_SQL Stage Update #203

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by LeonLuttenberger — APPROVED** [review]

**Review by malachi-constant — COMMENTED** [review]
Thanks ! @anmolsgandhi  I'd like to get a couple unit tests in here to make sure the new parameters are covered. I'm happy to help out with that today.

File: core/aws_ddk_core/stages/athena_sql.py [hunk 1]
```diff
@@ -65,8 +68,11 @@ def __init__(
             Identifier of the stage
         environment_id : str
             Identifier of the environment
-        query_string : str
+        query_string : Optional[str]
             SQL query that will be started
+        query_string_path : Optional[str]
+            dynamic path in statemachine for SQL query to be started
+            `$.queryString` by default
```
Comment (malachi-constant) [inline core/aws_ddk_core/stages/athena_sql.py:75]: I don't see this as a default any where in the codebase

File: core/aws_ddk_core/stages/athena_sql.py [hunk 2]
```diff
@@ -135,3 +150,16 @@ def __init__(
...
+    def get_targets(self) -> Optional[List[IRuleTarget]]:
+        return [
+            SfnStateMachine(
+                self.state_machine,
+                input=RuleTargetInput.from_object(self._state_machine_input),
+            )
+            if self._query_string
+            else SfnStateMachine(
+                self.state_machine,
+                input=RuleTargetInput.from_event_path(self._event_bridge_event_path),
```
Comment (kukushking) [inline core/aws_ddk_core/stages/athena_sql.py:163]: It looks like only the state machine input changes, could it be simplified into `SfnStateMachine` with a conditional `input=RuleTargetInput.from_object... if ... else RuleTargetInput.from_event_path...`?

**Review by kukushking — APPROVED** [review]
Awesome work!

File: core/aws_ddk_core/stages/athena_sql.py (hunk 2 above)
Comment (malachi-constant) [reply core/aws_ddk_core/stages/athena_sql.py:162]: +1

File: core/aws_ddk_core/stages/athena_sql.py (hunk 1 above)
Comment (anmolsgandhi) [reply core/aws_ddk_core/stages/athena_sql.py]: It was previously in my design but i decided against it, fixed this in code

**Review by malachi-constant — APPROVED** [review]

File: core/aws_ddk_core/stages/athena_sql.py (hunk 2 above)
Comment (anmolsgandhi) [reply core/aws_ddk_core/stages/athena_sql.py]: Thanks, Done

**Pull request merged**